zbuf/record.go

```diff
@@ -205,8 +205,8 @@ func (r *Record) TypeCheck() error {
 var (
 	ErrMissingField = errors.New("record missing a field")
 	ErrExtraField   = errors.New("record with extra field")
-	ErrNotContainer = errors.New("scalar where container was expected")
-	ErrNotScalar    = errors.New("container where scalar was expected")
+	ErrNotContainer = errors.New("primitive where container was expected")
+	ErrNotPrimitive = errors.New("container where primitive was expected")
```

Comment:
nit: these sound awkward to me.  Is "got (or saw? found?) primitive type where container was expected" (and vice versa) too wordy?

Reply:
i think i came up with these, if the proposed replacement is less awkward then sounds good to me.